mongo/integration/retryable_writes_prose_test.go

```diff
@@ -214,4 +217,76 @@ func TestRetryableWritesProse(t *testing.T) {
 				"expected an insert event, got a(n) %v event", cmdEvt.CommandName)
 		}
 	})
+
+	nwpCommandMonitor := new(event.CommandMonitor)
+	nwpClientOpts := options.Client().SetMonitor(nwpCommandMonitor).SetRetryWrites(true)
+
+	mtNWPOpts := mtest.NewOptions().ClientOptions(nwpClientOpts).MinServerVersion("6.0").
+		Topologies(mtest.ReplicaSet)
+
+	mt.RunOpts(fmt.Sprintf("%s label returns original error", driver.NoWritesPerformed), mtNWPOpts,
+		func(mt *mtest.T) {
+			const shutdownInProgressErrorCode int32 = 91
+			const notWritablePrimaryErrorCode int32 = 10107
+
+			// Disable any enabled fail points on exit.
+			defer mt.SetFailPoint(mtest.FailPoint{
+				ConfigureFailPoint: "failCommand",
+				Mode:               "off",
+			})
+
+			// Configure a fail point for a "ShutdownInProgress" error.
+			mt.SetFailPoint(mtest.FailPoint{
+				ConfigureFailPoint: "failCommand",
+				Mode:               mtest.FailPointMode{Times: 1},
+				Data: mtest.FailPointData{
+					WriteConcernError: &mtest.WriteConcernErrorData{
+						Code:        shutdownInProgressErrorCode,
+						ErrorLabels: &[]string{driver.RetryableWriteError},
+					},
+					FailCommands: []string{"insert"},
+				},
+			})
+
+			// secondFailPointConfigured is used to determine if the conditions from the
+			// shutdownInProgressErrorCode actually configures the "NoWritablePrimary" fail command.
+			var secondFailPointConfigured bool
+
+			// Set a command monitor on the client that configures a failpoint with a "ShutdownInProgress"
+			// label for a "NoWritablePrimary" error.
+			nwpCommandMonitor.Succeeded = func(_ context.Context, evt *event.CommandSucceededEvent) {
+				var errorCode int32
+				if wce := evt.Reply.Lookup("writeConcernError"); wce.Type == bsontype.EmbeddedDocument {
+					errorCode = wce.Document().Lookup("code").Int32()
+				}
+
+				// Do not set a fail point if event was not a writeConcernError with an error code for
+				// "ShutdownInProgress".
+				if errorCode != shutdownInProgressErrorCode {
+					return
+				}
+
+				mt.SetFailPoint(mtest.FailPoint{
+					ConfigureFailPoint: "failCommand",
+					Mode:               mtest.FailPointMode{Times: 1},
+					Data: mtest.FailPointData{
+						ErrorCode: notWritablePrimaryErrorCode,
+						ErrorLabels: &[]string{
+							driver.NoWritesPerformed,
+							driver.RetryableWriteError,
+						},
+						FailCommands: []string{"insert"},
+					},
+				})
+				secondFailPointConfigured = true
+			}
+
+			// Attempt to insert a record to any collection on any database using "InsertOne".
```

Comment:
```suggestion
			// Attempt to insert a document.
```
Not sure what you mean by `record` and this will insert to the collection created as part of mtest setup, which I believe has the same name as the test (so not "any collection on any database").